blog/2022-12-13-dhis-to-rapidpro-in-the-field.md

```diff
@@ -0,0 +1,177 @@
+
+```shell
+./dhis2rapidpro.jar --rapidpro.api.url=https://rapidpro.dhis2.org/api/v2 \
+ --dhis2.api.url=https://play.dhis2.org/2.39.0/api \
+ --server.ssl.enabled=false --server.port=8081 \
+ --rapidpro.flow.uuids=f23c4129-872b-464f-a1a2-afa89cdd9b82
+```
+
+The argument to focus on is `rapidpro.flow.uuids`. The application will scan for flow runs that belong to the flow definition `f23c4129-872b-464f-a1a2-afa89cdd9b82` as referenced in this argument. The flow definition ID was copied from the browser address bar while on the RapidPro flow’s designer page:
+
+![Flow designer](./assets/dhis-to-rapidpro-in-the-field/flow-designer.png)
+
+Successfully completed flow runs are transformed into data value sets and pushed into DHIS2. Any hiccups in the network communication between RapidPro and DHIS-to-RapidPro only _interrupts_ the latter from ingesting the report rather than losing the report itself. 
+
+Before we could start transferring cell phone transmitted reports to DHIS2, we manually assigned the contact sending the SMS to a DHIS2 organisation unit. After all, DHIS-to-RapidPro needs to know which village the contact is reporting for. DHIS-to-RapidPro supports the mapping of DHIS2 users to RapidPro contacts (i.e., `--sync.rapidpro.contacts=true`) which allows contacts to be automatically assigned to organisation units. However, in the context of this pilot, the contacts were not DHIS2 users so the contact under test had to be manually assigned to the organisation unit. To this end, from the RapidPro contact field management web page, we proceeded to create a contact field named `DHIS2 Organisation Unit ID`:
+
+![Create field dialog](./assets/dhis-to-rapidpro-in-the-field/create-field.png)
+
+Within the configuration page of the contact under test, this field was set to the identifier of an organisation unit that has access to the relevant data set:
+
+![Field value dialog](./assets/dhis-to-rapidpro-in-the-field/org-unit-code.png)
+
+DHIS-to-RapidPro picks the report’s organisation unit identifier from this field when delivering the contact’s report. You might have observed that the identifier `101010102` is a DHIS2 code and not an opaque DHIS2 ID. Zimbabwe’s MoHCC has a code system for identifying villages. Giving village codes to contacts will help MoHCC in the future assign contacts to organisation units. The argument to have DHIS-to-RapidPro handle the organisation unit identifiers as codes is `org.unit.id.scheme=code`:
```

Comment:
Looks like the identifier in the text is different than the image, are they supposed to be equal?